### PR TITLE
Personal Blog Templates: Text-only alternative

### PR DESCRIPTION
**Pull request by carolinan**
**Description**

Adds Personal Blog templates (Text-only alternative) listed in https://github.com/WordPress/twentytwentyfive/issues/132
This PR includes the file changes from https://github.com/WordPress/twentytwentyfive/pull/138, as patterns.

**Screenshots**
Home:
text-only-blog-home

text-only-blog-home-small

Archive:

text-only-blog-archive

Single:

text-only-blog-single

Search:
text-only-blog-search

**Testing Instructions**
Activate Gutenberg.
Apply the PR.
On your WordPress test installation, make sure that you have enough content to trigger the query pagination and the next and previous post navigation.

Go to Appearance > Editor and open the Home template to edit it.
In the "Template" panel in the sidebar, open the "Design" panel and select the "text-only blog, home" template.
Save.
Confirm that the design matches the design on Figma, both in the editor and front.

Now repeat the same test for the archive, search, and single post templates.

### CODE REVIEW

**Comment by carolinan** [comment]
The first commits copies the files from https://github.com/WordPress/twentytwentyfive/pull/138 into patterns.

Todo: 

- [x] Move the query pagination inside the query loop, because it does not work outside of it. When there are no search results for example, the query pagination still shows on the front.

**Comment by carolinan** [comment]
[Comparing to the design in Figma,](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3142-472&m=dev) I believe the following needs adjusting:

- [ ] - Increase spacing between the archive description and the list of posts
- [ ] - Increase spacing between the search form and the list of posts
- [ ] - Decrease the font size of the search form from 22 to 18.

File: patterns/text-only-blog-home-template.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:template-part {"slug":"header","tagName":"header","area":"header"} /-->
+
+<!-- wp:group {"tagName":"main","style":{"spacing":{"margin":{"top":"var:preset|spacing|60"}}},"layout":{"type":"constrained"}} -->
+<main class="wp-block-group" style="margin-top:var(--wp--preset--spacing--60)">
+	<!-- wp:heading {"level":1,"align":"wide","fontSize":"large"} -->
+	<h1 class="wp-block-heading alignwide has-large-font-size"><?php esc_html_e( 'Blog', 'twentytwentyfive' ); ?></h1>
```
Comment (juanfra) [inline patterns/text-only-blog-home-template.php]: I don't see this in the design, but I assume it's to have a proper hierarchy? 

What I noticed is that it is the same size as the title of the blog posts.

File: patterns/text-only-blog-posts.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			<p>No posts were found.</p>
+			<!-- /wp:paragraph -->
+		<!-- /wp:query-no-results -->
+	</div>
+	<!-- /wp:group -->
+	<!-- wp:post-template {"align":"full","style":{"spacing":{"blockGap":"0"}},"layout":{"type":"default"}} -->
+		<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"var:preset|spacing|30","bottom":"var:preset|spacing|30"}},"border":{"bottom":{"color":"var:preset|color|opacity-20","width":"1px"},"top":{},"right":{},"left":{}}},"layout":{"type":"flex","flexWrap":"nowrap","verticalAlignment":"center","justifyContent":"space-between"}} -->
+		<div class="wp-block-group alignfull" style="border-bottom-color:var(--wp--preset--color--opacity-20);border-bottom-width:1px;padding-top:var(--wp--preset--spacing--30);padding-bottom:var(--wp--preset--spacing--30)">
+			<!-- wp:post-title {"isLink":true,"fontSize":"large"} /-->
+			<!-- wp:post-date {"textAlign":"right","isLink":true,"fontSize":"small"} /-->
```
Comment (juanfra) [inline patterns/text-only-blog-posts.php:28]: I noticed this one is set to `12px` in the design, but we don't have such a preset, as the lower is now `14px`.

@beafialho we may want to update the figma to reflect this.

File: patterns/text-only-blog-posts.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+		<!-- /wp:group -->
+	<!-- /wp:post-template -->
+	
+	<!-- wp:spacer {"height":"var:preset|spacing|30"} -->
+	<div style="height:var(--wp--preset--spacing--30)" aria-hidden="true" class="wp-block-spacer"></div>
+	<!-- /wp:spacer -->
+
+	<!-- wp:group {"align":"full","style":{"spacing":{"margin":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40"}}},"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group alignfull" style="margin-top:var(--wp--preset--spacing--40);margin-bottom:var(--wp--preset--spacing--40);">
+		<!-- wp:query-pagination {"align":"full","style":{"typography":{"fontStyle":"normal","fontWeight":"400"}},"layout":{"type":"flex","justifyContent":"space-between","flexWrap":"wrap"}} -->
```
Comment (juanfra) [inline patterns/text-only-blog-posts.php:39]: I believe that now that we have changed the presets, normal will be smaller than it was, and to set it to "normal," we'll have to use a large font size (22px). We may also want to do it as a `theme.json` level.

File: patterns/text-only-blog-single-post-template.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+?>
+<!-- wp:template-part {"slug":"header","tagName":"header","area":"header"} /-->
+
+<!-- wp:group {"tagName":"main","style":{"spacing":{"margin":{"top":"var:preset|spacing|60"}}},"layout":{"type":"constrained"}} -->
+<main class="wp-block-group" style="margin-top:var(--wp--preset--spacing--60)">
+	<!-- wp:group {"align":"full","style":{"spacing":{"padding":{"top":"var:preset|spacing|60","bottom":"var:preset|spacing|60"}}},"layout":{"type":"constrained"}} -->
+	<div class="wp-block-group alignfull" style="padding-top:var(--wp--preset--spacing--60);padding-bottom:var(--wp--preset--spacing--60)">
+		<!-- wp:post-terms {"term":"category","className":"is-style-default","style":{"typography":{"fontStyle":"normal","fontWeight":"400"}}} /-->
+		<!-- wp:post-title {"level":1,"style":{"spacing":{"margin":{"bottom":"var:preset|spacing|50"}}},"fontSize":"xx-large"} /-->
```
Comment (juanfra) [inline patterns/text-only-blog-single-post-template.php:22]: It looks in the designs that the top and the bottom spacing should be `preset|spacing|50` (`30px`).

File: patterns/text-only-blog-single-post-template.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+		<div class="wp-block-group alignfull" style="margin-top:var(--wp--preset--spacing--60);margin-bottom:var(--wp--preset--spacing--60);padding-right:var(--wp--preset--spacing--50);padding-left:var(--wp--preset--spacing--50)">
+			<!-- wp:group {"tagName":"nav","align":"wide","style":{"border":{"top":{"color":"var:preset|color|opacity-20","width":"1px"},"right":[],"bottom":[],"left":[]},"spacing":{"padding":{"top":"var:preset|spacing|40","bottom":"var:preset|spacing|40"}}},"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"space-between"}} -->
+			<nav class="wp-block-group alignwide" aria-label="Posts navigation" style="border-top-color:var(--wp--preset--color--opacity-20);border-top-width:1px;padding-top:var(--wp--preset--spacing--40);padding-bottom:var(--wp--preset--spacing--40)">
+				<!-- wp:post-navigation-link {"type":"previous","showTitle":true,"arrow":"arrow"} /-->
+				<!-- wp:post-navigation-link {"showTitle":true,"arrow":"arrow"} /-->
```
Comment (juanfra) [inline patterns/text-only-blog-single-post-template.php:37]: For these two, the design shows a font size of `18px` and locally `14px`.

**Review by juanfra — COMMENTED** [review]
Thanks, Carolina! Code looks good. I noticed a couple mismatches with the design that we should probably review with Bea.

**Comment by beafialho** [comment]
Thank you for turning the templates into patterns.

@juanfra please do not look at the Figma file to address this one (neither the all the othertemplates I build), since I built them in the editor and made decisions accordingly there. I will review how they look based on the how the changes look in my testing.

I noticed a few things not looking as intended, likely due to the change in the font presets (this change will make reviewing designs much more confusing since this is not represented in the Figma) for others, that's why I would prefer reviewing case by case.

- I'm seeing the headings (Blog, Archive title, Search results title) are using the L preset font size, and they should all be using the XL preset, so that they are not the same size as the blog posts.
- I'm also seeing the pagination using the S font size preset, it should use the M preset.

> Increase spacing between the archive description and the list of posts
> Increase spacing between the search form and the list of posts

- Lastly, you're correct @carolinan, I'm currently seeing spacing preset X-Small. We can apply the spacers between the titles and post lists to use the Regular spacing preset in both Archive, Search results and Blog templates.

Please let me know if that was clear.

**Comment by carolinan** [comment]
This is all so confusing but I know we will work it out somehow.
- I updated the slugs for the font sizes one hour ago, so there should not be any x-small left, not for the fonts at least.

**Comment by carolinan** [comment]
Aha! The PR to change the font size slugs missed the query pagination.

**Comment by carolinan** [comment]
>Lastly, you're correct @carolinan, I'm currently seeing spacing preset X-Small. We can apply the spacers between the titles and post lists to use the Regular spacing preset in both Archive, Search results and Blog templates.
>Please let me know if that was clear.

I am only seeing spacers **inside** the query, so I will add new ones between the titles and posts lists.

**Comment by carolinan** [comment]
In this screenshot, the spacer between the search and the post list uses regular:
search-results-page

In this screenshot, the spacer uses small:
search-results-page-small

Side by side comparison, regular on left, small on right
image

**Comment by juanfra** [comment]
> @juanfra please do not look at the Figma file to address this one (neither the all the othertemplates I build), since I built them in the editor and made decisions accordingly there. I will review how they look based on the how the changes look in my testing.

@beafialho Ok, should I do this for only templates? I'm a bit confused about what to highlight and what not to when I review. 

I share the inconsistencies I catch, as I believe the more we catch now, the less of complications we'll have further down the road.

**Comment by beafialho** [comment]
Yes @juanfra, only for these PRs I submitted with templates. For everything else, please continue taking the designs on Figma.

> I share the inconsistencies I catch, as I believe the more we catch now, the less of complications we'll have further down the road.
I totally understand. Thank you for your thoroughness.

File: patterns/text-only-blog-single-post-template.php (hunk 5 above)
Comment (carolinan) [reply patterns/text-only-blog-single-post-template.php:37]: I made the size change to the block in theme.json.

**Comment by beafialho** [comment]
Just making sure, because I was just now seeing two spacers, we're only using **one spacer** using the Regular size preset, right? Like this:

https://github.com/user-attachments/assets/d692813e-2397-4670-b47b-8af2b4fc6b2b

If this is it, we're good to merge 🎉

**Comment by carolinan** [comment]
There are two spacers.

The spacer that is in this video, that is inside the no results block, only shows when there are no results.
That spacer is between the  archive title and the "No posts were found." message.

The newly added spacer is between the post lists and:

- The term description, if there is one, otherwise the archive title.
- The search form.
- The heading that says "Blog".

**Comment by carolinan** [comment]
> The spacer that is in this video, that is inside the no results block, only shows when there are no results. That spacer is between the archive title and the "No posts were found." message.

Shall I remove this one?

**Comment by beafialho** [comment]
> Shall I remove this one?

Yes 👍 thank you

**Pull request merged**